react/features/base/redux/middleware.js

```diff
@@ -0,0 +1,54 @@
+/* @flow */
+import Logger from 'jitsi-meet-logger';
+
+import { persistState } from './functions';
+import MiddlewareRegistry from './MiddlewareRegistry';
+
+const logger = Logger.getLogger(__filename);
+
+/**
+ * The delay that passes between the last state change and the state to be
+ * persisted in the storage.
+ */
+const PERSIST_DELAY = 2000;
+
+/**
+ * The timeout ID that the throttler uses.
+ */
+let _throtthleTimeoutID;
+
+/**
+ * A master MiddleWare to selectively persist state. Please use the
+ * {@link persisterconfig.json} to set which subtrees of the Redux state
+ * should be persisted.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(store => next => action => {
+    const result = next(action);
+
+    _throttledPersist(store.getState());
```

Comment:
A couple notes here: 
1. Technically this is a debounce, I think. Underscore debounce could be used, even, if it does what you want.
2. Be sure to use the wrapper for accessing local storage. That way it's easier to stub out as needed for when it's not available and when implementation needs to change (and for the unit tests I can dream about).

One note is that initially I thought this function persisted the entire stores state. I've basically gotten used to ignoring all the docs now. But now that I know about the config, it's evident to me how it works. I'm wondering if it makes any sense to filter outside of persistState so that persistState gets passed in only what it needs instead or persist state to reference that it filters.

Reply:
Good point about debounce, I didn't know that it's in the app. I'll use that.
What do you mean about local storage wrapper?
Also, I'm not sure about filtering the state before it gets passed to the function because this way the function encapsulates all the config usage and everything, while if I filter it prior to passing it then this logic moves over to the middleware (if I understand you correctly?). Also, this way anyone can use the "persistState" function outside of the feature because that encapsulates all functionality that is needed. What do you think?

Reply:
Actually I used "throttle", that is a bit more like what we need here.

Reply:
Web has a wrapper around local storage. Maybe mobile doesn't?

Reply:
We talked about this offline and mobile doesn't have a synchronous local storage wrapper and maybe one will be made in the future.